### PR TITLE
Removed unnecessary null checking

### PR DESCRIPTION
**Pull request by Finka95**
- Removed unnecessary null checking by changing the entity search method in TicketService and UserService
- Replase AsNoTracking method in GenericRepository

### CODE REVIEW

File: TicketSystem/TicketSystem.BLL/Services/TicketService.cs [hunk 1]
```diff
@@ -62,13 +62,11 @@ public async Task<IEnumerable<Ticket>> GetTicketsAsync(CancellationToken cancell
         return _mapper.Map<IEnumerable<Ticket>>(ticketsEntity);
     }
 
-    public async Task<Ticket> UpdateTicketAsync(Ticket ticketModel, CancellationToken cancellationToken)
+    public async Task<Ticket> UpdateTicketAsync(Ticket ticket, CancellationToken cancellationToken)
     {
-        var ticketEntity =
-            await _ticketRepository.GetByIdWithIncludeAsync(ticketModel.Id, cancellationToken);
+        await GetTicketByIdAsync(ticket.Id, cancellationToken);
```
Comment (nantonov) [inline TicketSystem/TicketSystem.BLL/Services/TicketService.cs]: Why GetTicketByIdAsync doesn't return the ticket? Looks strange

File: TicketSystem/TicketSystem.BLL/Services/TicketService.cs [hunk 2]
```diff
@@ -77,9 +75,9 @@ public async Task<Ticket> UpdateTicketAsync(Ticket ticketModel, CancellationToke
 
     public async Task<Ticket> DeleteTicketAsync(int id, CancellationToken cancellationToken)
     {
-        var ticketEntity = await _ticketRepository.GetByIdWithIncludeAsync(id, cancellationToken);
-        if (ticketEntity == null)
-            throw new NotFoundException($"Ticket with id {id} not found");
+        var ticket = await GetTicketByIdAsync(id, cancellationToken);
```
Comment (nantonov) [inline TicketSystem/TicketSystem.BLL/Services/TicketService.cs]: We can delete by id, without retrieving ticket

**Review by nantonov — CHANGES_REQUESTED** [review]

File: TicketSystem/TicketSystem.DAL/Repositories/GenericRepository.cs [hunk 3]
```diff
@@ -28,8 +29,10 @@ public async Task UpdateAsync(TEntity item, CancellationToken cancellationToken)
         await _context.SaveChangesAsync(cancellationToken);
     }
 
-    public async Task RemoveAsync(TEntity item, CancellationToken cancellationToken)
+    public async Task RemoveAsync(int id, CancellationToken cancellationToken)
     {
+        var item = await _dbSet.FirstAsync(x => x.Id == id, cancellationToken: cancellationToken);
```
Comment (nantonov) [inline TicketSystem/TicketSystem.DAL/Repositories/GenericRepository.cs]: FirstOrDefault will be better 
If we already don't have this resource - good news, we already done this job :)

File: TicketSystem/TicketSystem.DAL/Repositories/GenericRepository.cs [hunk 4]
```diff
@@ -47,15 +50,15 @@ public async Task RemoveAsync(TEntity item, CancellationToken cancellationToken)
...
         params Expression<Func<TEntity, object>>[] includeProperties)
     {
-        var entity = _dbSet.AsNoTracking();
+        IQueryable<TEntity> entity = _dbSet;
 
         if (includeProperties.Length > 0)
             entity = Include(includeProperties);
 
-        return await entity.FirstOrDefaultAsync(t => t.Id == id, cancellationToken);
+        return await entity.AsNoTracking().FirstAsync(t => t.Id == id, cancellationToken);
```
Comment (nantonov) [inline TicketSystem/TicketSystem.DAL/Repositories/GenericRepository.cs]: It's okay to return null

File: TicketSystem/TicketSystem/Middleware/ExceptionMiddleware.cs [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ﻿using System.Net;
 using TicketSystem.API.ViewModels.Exceptions;
-using TicketSystem.BLL.Exceptions;
+using TicketSystem.DAL.Exceptions;
```
Comment (nantonov) [inline TicketSystem/TicketSystem/Middleware/ExceptionMiddleware.cs]: API cannot use DAL

**Review by nantonov — CHANGES_REQUESTED** [review]

**Review by nantonov — APPROVED** [review]

**Review by 5Ka-me — APPROVED** [review]

**Review by moltenessence — APPROVED** [review]

**Review by Genresal — APPROVED** [review]

**Review by Genresal — APPROVED** [review]

**Pull request merged**